builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -148,6 +153,17 @@ func resourceCloudStackInstanceCreate(d *schema.ResourceData, meta interface{})
 		p.SetUserdata(ud)
 	}
 
+	// If the project_name contains any info, we retreive the project_id
+	if projectName, ok := d.GetOk("project_name"); ok {
+		project, _, err := cs.Project.GetProjectByName(projectName.(string))
+		if err != nil {
+			return err
+		}
+		log.Printf("[DEBUG] project id %s", project.Id)
+		p.SetProjectid(project.Id)
+		d.Set("project_id", project.Id)
```

Comment:
You can (and should) delete this line. You are trying to set a value for the parameter `project_id`, but that parameter actually doesn't exist (it's not specified in the schema). So even if this doesn't trigger an error (not sure about that), the value will not be saved.